### PR TITLE
Document System.Runtime.AmbiguousImplementationException

### PR DESCRIPTION
**Pull request by carlossanlop**
Here's the original proposal for the exception: https://github.com/dotnet/corefx/issues/34124

Here's the PR that introduced it: https://github.com/dotnet/coreclr/pull/22280

Adding @MichalStrehovsky who proposed the API and @JKotas who approved it.

### CODE REVIEW

**Comment by carlossanlop** [comment]
CC @joperezr and @bartonjs, who are marked as owners of System.Runtime, but I can see this area is huge. 😃

File: xml/System.Runtime/AmbiguousImplementationException.xml [hunk 1]
```diff
@@ -14,7 +14,7 @@
   </Base>
   <Interfaces />
   <Docs>
-    <summary>To be added.</summary>
+    <summary>The exception that is thrown when there is an ambiguous implementation causing diamond inheritance situation. This situation happens when there are multiple incompatible interface methods overriding another method.</summary>
```
Comment (bartonjs) [inline xml/System.Runtime/AmbiguousImplementationException.xml]:
```suggestion
    <summary>The exception that is thrown when there are multiple incompatible interface methods overriding another method.</summary>
```

File: xml/System.Runtime/AmbiguousImplementationException.xml [hunk 2]
```diff
@@ -51,8 +51,8 @@
         <Parameter Name="message" Type="System.String" />
       </Parameters>
       <Docs>
-        <param name="message">To be added.</param>
-        <summary>To be added.</summary>
+        <param name="message">The localized error message string.</param>
```
Comment (bartonjs) [inline xml/System.Runtime/AmbiguousImplementationException.xml]:
```suggestion
        <param name="message">A <see cref="System.String"/> that describes the error.</param>
```

(Borrowed from https://docs.microsoft.com/en-us/dotnet/api/system.missingmethodexception.-ctor?view=netframework-4.8#System_MissingMethodException__ctor_System_String_)

File: xml/System.Runtime/AmbiguousImplementationException.xml [hunk 3]
```diff
@@ -73,9 +73,9 @@
         <Parameter Name="innerException" Type="System.Exception" />
       </Parameters>
       <Docs>
-        <param name="message">To be added.</param>
-        <param name="innerException">To be added.</param>
-        <summary>To be added.</summary>
+        <param name="message">The localized error message string.</param>
+        <param name="innerException">The exception that is the cause of the current exception. If the <paramref name="innerException" /> parameter is not a null reference (<see langword="Nothing" /> in Visual Basic), the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</param>
```
Comment (bartonjs) [inline xml/System.Runtime/AmbiguousImplementationException.xml]: @jkotas Is the deferred catch here true?  MissingMethodException also says it, but I'm not familiar with an inner exception being a candidate for a catch, and a thrown InvalidOperationException inside a manually thrown MissingMethodException certainly didn't trigger `catch (InvalidOperationException)` in a test just now.

Seems like this should just be

```suggestion
        <param name="innerException">The exception that is the cause of the current exception.</param>
```

**Comment by MichalStrehovsky** [comment]
Do we want to add a remarks section? Something along the lines of:

> Normally a compilation error is generated if there are multiple incompatible interface methods implementing another interface method. AmbiguousImplementationException is thrown when a type in a dependent assembly implements an interface from an assembly that was modified.
> 
> To resolve the error, a most specific implementation of the interface method has to be provided. Most specific implementation of the interface method can be provided on the class that implements the interface, or on another interface that _requires_ the interface providing the conflicting implementation.
> 
> AmbiguousImplementationException uses the HRESULT COR_E_AMBIGUOUSIMPLEMENTATION, that has the value 0x8013106A.

(Took the inspiration from MissingMethodException.)

File: xml/System.Runtime/AmbiguousImplementationException.xml (hunk 2 above)
Comment (carlossanlop) [reply xml/System.Runtime/AmbiguousImplementationException.xml]: From recent experience, it seems that the Docs team prefers to keep simple messages like this one in plain english if possible, without adding crefs.

**Comment by carlossanlop** [comment]
@MichalStrehovsky I'd be glad to add your remark message. Do you want me to add it to the general `<remarks>` element, or to a specific constructor?

File: xml/System.Runtime/AmbiguousImplementationException.xml [hunk 4]
```diff
@@ -14,7 +14,7 @@
   </Base>
   <Interfaces />
   <Docs>
-    <summary>To be added.</summary>
+    <summary>The exception that is thrown when there are multiple incompatible interface methods overriding another method.</summary>
     <remarks>To be added.</remarks>
```
Comment (carlossanlop) [inline xml/System.Runtime/AmbiguousImplementationException.xml]: @MichalStrehovsky I assume you wanted it here, in the general remarks section:

```suggestion
    <remarks>
      <format type="text/markdown"><![CDATA[

## Remarks

Normally, a compilation error is generated if there are multiple incompatible interface methods implementing another interface method. `AmbiguousImplementationException` is thrown when a type in a dependent assembly implements an interface from an assembly that was modified.

To resolve the error, a most specific implementation of the interface method has to be provided. The most specific implementation of the interface method can be provided on the class that implements the interface, or on another interface that requires the interface providing the conflicting implementation.

`AmbiguousImplementationException` uses the `HRESULT COR_E_AMBIGUOUSIMPLEMENTATION`, which has a value of `0x8013106A`.

      ]]></format>
    </remarks>
```

File: xml/System.Runtime/AmbiguousImplementationException.xml (hunk 2 above)
Comment (bartonjs) [reply xml/System.Runtime/AmbiguousImplementationException.xml]: Fair enough, but there's no requirement that it be localized, which is mainly what I was editing out.

File: xml/System.Runtime/AmbiguousImplementationException.xml [hunk 5]
```diff
@@ -14,8 +14,20 @@
...
-    <remarks>To be added.</remarks>
+    <summary>The exception that is thrown when there are multiple incompatible interface methods overriding another method.</summary>
+    <remarks>
+      <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+Normally, a compilation error is generated if there are multiple incompatible interface methods implementing another interface method. `AmbiguousImplementationException` is thrown when a type in a dependent assembly implements an interface from an assembly that was modified.
+
+To resolve the error, a most specific implementation of the interface method has to be provided. The most specific implementation of the interface method can be provided on the class that implements the interface, or on another interface that requires the interface providing the conflicting implementation.
```
Comment (rpetrusha) [inline xml/System.Runtime/AmbiguousImplementationException.xml]:
```suggestion
To resolve the error, the most specific implementation of the interface method has to be provided. The most specific implementation of the interface method can be provided on the class that implements the interface, or on another interface that requires the interface providing the conflicting implementation.
```

File: xml/System.Runtime/AmbiguousImplementationException.xml (hunk 2 above)
Comment (rpetrusha) [reply xml/System.Runtime/AmbiguousImplementationException.xml]: I've eliminated the cref and simply changed it to "string".

File: xml/System.Runtime/AmbiguousImplementationException.xml [hunk 6]
```diff
@@ -73,11 +85,11 @@
         <Parameter Name="innerException" Type="System.Exception" />
       </Parameters>
       <Docs>
-        <param name="message">To be added.</param>
-        <param name="innerException">To be added.</param>
-        <summary>To be added.</summary>
+        <param name="message">The localized error message string.</param>
```
Comment (rpetrusha) [inline xml/System.Runtime/AmbiguousImplementationException.xml]: Should this also be "A string that describes the error."?

Comment (carlossanlop) [reply xml/System.Runtime/AmbiguousImplementationException.xml]: Sure, let's do the same:

```suggestion
        <param name="message">A string that describes the error.</param>
```

File: xml/System.Runtime/AmbiguousImplementationException.xml (hunk 2 above)
Comment (carlossanlop) [reply xml/System.Runtime/AmbiguousImplementationException.xml]: Ah I see what you mean, Jeremy, thanks for catching that.

**Comment by carlossanlop** [comment]
@rpetrusha comments addressed, build passed. Is it good to merge?

**Comment by rpetrusha** [comment]
Yes, it's good to merge, @carlossanlop. I'll do that now.
